### PR TITLE
P2 Shopping Cart UI

### PR DESCRIPTION
**Pull request by djoyal**
This PR adds UI elements to support showing cart information on the P2 page. Specifically:
* The number of each menu item currently in the cart will be shown on each menu item
* The total number of items in the cart will be shown on the 'View Cart' button

**Example Screenshot**
Screen Shot 2021-01-27 at 4 20 27 PM

### CODE REVIEW

File: packages/web/components/itemSummary.tsx [hunk 1]
```diff
@@ -1,45 +1,53 @@
...
+        pathname: '/establishment/[establishmentId]/menu/[id]',
         // The next line contains some serious trickery. Strap yourself in.
         // The 'router.query' objects contains all of the parameters in the curent URL.
         // So, if you're currently on route `/establishment/1`, router.query will contain
         // establishmentId: 1. Using the spread operator (...), we can inject those values
         // back into the URL we're building using the format from 'pathname' above.
-        query: { ...router.query, id: props.id },
-      }}
-    >
+        query: { ...router.query, id: id },
```
Comment (jfplataroti) [inline packages/web/components/itemSummary.tsx:25]: nice!!!

File: packages/web/components/itemSummary.tsx [hunk 2]
```diff
@@ -1,45 +1,53 @@
...
+      }}>
       <Box direction="column" width="medium" pad="xsmall">
         <Box background="#F0F0F0">
-          <Image
-            src={props.image}
-            style={{ borderRadius: "8px" }}
-            fill="horizontal"
-          />
+          <Stack anchor="top-left">
```
Comment (jfplataroti) [inline packages/web/components/itemSummary.tsx:29]: nice use of grommet with the `anchor` attribute.

File: packages/web/pages/establishment/[establishmentId]/menu/[id].tsx [hunk 3]
```diff
@@ -63,28 +79,28 @@ export default function MenuItem({
         <Heading>{title}</Heading>
         <Image
           src={GetImagePath(media.image.path)}
-          style={{ borderRadius: "8px" }}
+          style={{ borderRadius: '8px' }}
```
Comment (jfplataroti) [inline packages/web/pages/establishment/[establishmentId]/menu/[id].tsx:82]: Maybe we can add the `borderRadius` to the `theme` in the global section. I remember setting the same value somewhere.

File: packages/web/pages/establishment/[establishmentId]/menu/[id].tsx [hunk 4]
```diff
@@ -2,32 +2,36 @@ import {
...
+    subTitle,
+    allergens,
+    heroIngredients = [],
+    healthBenefits = [],
+    modifierGroups = [],
+    ingredientCategories = [],
+  } = props.item;
   const { t } = useTranslation();
+  const router = useRouter();
   const [modifiers, setModifiers] = useState(
```
Comment (jfplataroti) [inline packages/web/pages/establishment/[establishmentId]/menu/[id].tsx:35]: Hey sorry, I know this is not part of the PR, it's just something to think about. the initialization of the `useState` has a few lines to transform a model, what do you think about moving things like this transformation (when they are big) outside the component into its own function.

No need to be changed, just something to talk about :).

**Review by jfplataroti — APPROVED** [review]
I just tested locally and looks amazing!! :) 
Is starting to feel like the mockup, nice job! :)

File: packages/web/pages/establishment/[establishmentId]/menu/[id].tsx (hunk 4 above)
Comment (djoyal) [reply packages/web/pages/establishment/[establishmentId]/menu/[id].tsx:35]: That's a great suggestion! You are absolutely correct that this kind of logic should go into some kind of mapping function that lives outside of the components.

File: packages/web/pages/establishment/[establishmentId]/menu/[id].tsx (hunk 3 above)
Comment (djoyal) [reply packages/web/pages/establishment/[establishmentId]/menu/[id].tsx:82]: When I looked into this I realized two things:
1. In this specific case we don't need to use `borderRadius`
2. In general we can rely on the `grommet` border radius values by wrapping `Image` in a `Box` and setting the `round` property on the `Box`

Comment (jfplataroti) [reply packages/web/pages/establishment/[establishmentId]/menu/[id].tsx]: ah nice!!

File: packages/web/components/itemSummary.tsx [hunk 5]
```diff
@@ -25,13 +25,9 @@ export default function ItemSummary(props) {
         query: { ...router.query, id: id },
       }}>
       <Box direction="column" width="medium" pad="xsmall">
-        <Box background="#F0F0F0">
+        <Box background="#F0F0F0" round="medium">
```
Comment (jfplataroti) [inline packages/web/components/itemSummary.tsx:28]: 👍

File: packages/web/pages/establishment/[establishmentId]/menu/[id].tsx [hunk 6]
```diff
@@ -32,19 +33,7 @@ export default function MenuItem(props) {
...
-        modifiers: group.modifiers.map((modifier) => {
-          return {
-            ...modifier,
-            selected: modifier.quantity.included > 0 ? 1 : 0,
-          };
-        }),
-      };
-    }),
-  );
+  const [modifiers, setModifiers] = useSelectableModifiers(modifierGroups);
```
Comment (jfplataroti) [inline packages/web/pages/establishment/[establishmentId]/menu/[id].tsx:36]: 👍 great idea using a hook, will give us more options if we want to enhance it and control how ofter it's requested.

**Review by jfplataroti — APPROVED** [review]
thanks for looking into those two features. Grommet is exceeding my expectations hehe, and the custom hook for the modifiers it's a good call.

**Pull request merged**